encrypt.go

```diff
@@ -167,6 +189,38 @@ func (e *EncryptManager) encryptCFB(r io.Reader) ([]byte, error) {
 	return encrypted, nil
 }
 
+//encryptRSA encrypts given io.Reader using RSA-PCKS
+// the resultant encrypted bytes is returned
+func (e *EncryptManager) encryptRSA(r io.Reader) ([]byte, error) {
+	if r == nil {
+		return nil, errors.New("invalid content provided")
+	}
+
+	b, err := ioutil.ReadAll(r)
+	if err != nil {
+		return nil, err
+	}
+
+	rsaKeyPair, err := e.unmarshallRsaKey()
+
+	if err != nil {
+		return nil, err
+	}
+
+	if rsaKeyPair.pubkey.Size() < len(b) {
+		return nil, fmt.Errorf("Can't encrypt file larger than RSA pub key size")
```

Comment:
Ref -: Go Doc,  https://golang.org/pkg/crypto/rsa/#EncryptPKCS1v15

I have also noticed that its implementation already checks this condition [here](https://golang.org/src/crypto/rsa/pkcs1v15.go?s=1254:1334#L29), so I removed this from here.

Reply:
I see. I think this function is not intended to encrypt content directly but used to encrypt a symmetrical key that encrypts the content.

Reply:
> I think this function is not intended to encrypt content directly

Hmm,  Agreed that it's not recommended to use `PCKS` to encrypt the plaintext.

> but used to encrypt a symmetrical key that encrypts the content.

I am not sure, we will be doing symmetric key encryption with RSA keys?

Anyways, I took a more in-depth look about RSA encryption and came across [this](https://security.stackexchange.com/questions/183179/what-is-rsa-oaep-rsa-pss-in-simple-terms/183330#183330), so it seems like `PCKS Padding` is sorta outdated padding scheme that doesn't sit well with plaintext.

To encrypt plaintext optimal choice would be to use `OAEP Padding` scheme, I am gonna change the implementation to use it.

Reply:
"RSA is a relatively slow algorithm, and because of this, it is less commonly used to directly encrypt user data. More often, RSA passes encrypted shared keys for symmetric key cryptography which in turn can perform bulk encryption-decryption operations at much higher speed. " https://en.wikipedia.org/wiki/RSA_(cryptosystem)

"The message must be no longer than the length of the public modulus minus twice the hash length, minus a further 2. " https://golang.org/pkg/crypto/rsa/#EncryptOAEP

So to be able to encrypt large files, we need to combine RSA with a symmetric cipher.

Reply:
Correct, I think we can use existing implementation of Symmetric cyphers (i.e. GCM or CFB we have in this library).
 
> So to be able to encrypt large files, we need to combine RSA with a symmetric cipher.

The new process will be as follows -:
- For Encryption -: Input `I` will be the file (data that needs to be encrypted), a paraphrase `P` will be randomly generated, Data `I` will be encrypted using AES ciphers with paraphrase `P`, `P` will be encrypted using RSA Keystore (IPFS-keys). 
- For Decryption -: Encrypted input `I'` & encrypted paraphrase `P'` will be provided for decryption.